### PR TITLE
TW-92: Update logic and new design for set up homeserver

### PR DESCRIPTION
**Pull request by nqhhdev**
- Issue: #92 

- Resolved: 
 
https://github.com/linagora/twake-on-matrix/assets/99852347/2a109950-b4f9-454a-a77e-8b643a767657

https://github.com/linagora/twake-on-matrix/assets/99852347/f48cda70-9db0-471c-a559-97def1669f14

### CODE REVIEW

**Review by dab246 — APPROVED** [review]
lgtm

File: lib/pages/homeserver_picker/homeserver_picker.dart [hunk 1]
```diff
@@ -158,14 +157,24 @@ class HomeserverPickerController extends State<HomeserverPicker> with ConnectPag
...
-      if (!ssoSupported && matrix.loginRegistrationSupported == false) {
-        // Server does not support SSO or registration. We can skip to login page:
-        VRouter.of(context).to('login');
-      } else if (ssoSupported && matrix.loginRegistrationSupported == false) {
+      if (ssoSupported && matrix.loginRegistrationSupported == false) {
         setState(() {
           state = HomeserverState.ssoLoginServer;
           FocusManager.instance.primaryFocus?.unfocus();
         });
+        Map<String, dynamic>? rawLoginTypes;
```
Comment (drminh2807) [inline lib/pages/homeserver_picker/homeserver_picker.dart:165]:
```suggestion
        final rawLoginTypes = await Matrix.of(context)
```

File: lib/pages/homeserver_picker/homeserver_picker_view.dart [hunk 2]
```diff
@@ -129,14 +118,13 @@ class HomeserverPickerView extends StatelessWidget {
     );
     switch (controller.state) {
       case HomeserverState.otherLoginMethod:
-      case HomeserverState.enterServerName:
-        return Text(L10n.of(context)!.letsStart, style: textStyle,);
+        return Text(L10n.of(context)!.loginWithSSO, style: textStyle,);
```
Comment (drminh2807) [inline lib/pages/homeserver_picker/homeserver_picker_view.dart:121]:
```suggestion
      case HomeserverState.otherLoginMethod:
      case HomeserverState.ssoLoginServer:
      case HomeserverState.wrongServerName:
        return Text(L10n.of(context)!.loginWithSSO, style: textStyle,);
```

**Review by drminh2807 — CHANGES_REQUESTED** [review]

File: lib/pages/homeserver_picker/homeserver_picker.dart [hunk 3]
```diff
@@ -34,7 +34,7 @@ class HomeserverPicker extends StatefulWidget {
 
 class HomeserverPickerController extends State<HomeserverPicker> with ConnectPageMixin {
   
-  HomeserverState state = HomeserverState.enterServerName;
+  HomeserverState state = HomeserverState.ssoLoginServer;
```
Comment (hoangdat) [inline lib/pages/homeserver_picker/homeserver_picker.dart:37]: please note that, we also support 3rd party server. Agree, we can bypass the unnecessary click for user when `sso` support, but only bypass when we know this server accepts SSO

File: lib/pages/homeserver_picker/homeserver_picker.dart (hunk 1 above)
Comment (nqhhdev) [reply lib/pages/homeserver_picker/homeserver_picker.dart:165]: You wrongly, `rawLoginTypes` is response of `Matrix.of(context).getLoginClient().request...`

File: lib/pages/homeserver_picker/homeserver_picker.dart (hunk 3 above)
Comment (nqhhdev) [reply lib/pages/homeserver_picker/homeserver_picker.dart:37]: Updated, pls check demo

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
please rebase and re-test with `matrix.org` also

**Comment by nqhhdev** [comment]
> please rebase and re-test with `matrix.org` also

Demo for matrix.org

https://github.com/linagora/twake-on-matrix/assets/99852347/ed1e883c-bb76-48f4-ad26-4a4c58d71591

**Pull request merged**